commit bc1cfd1f46caf7bf9cfdc55bd637144b45c2a0b2
Author: looooo <sppedflyer@gmail.com>
Date:   Thu Jan 31 11:19:21 2019 +0100

    patch2

diff --git a/Rendering/Matplotlib/module.cmake b/Rendering/Matplotlib/module.cmake
index 19ecf199..3728f5e6 100644
--- a/Rendering/Matplotlib/module.cmake
+++ b/Rendering/Matplotlib/module.cmake
@@ -26,7 +26,9 @@ vtk_module(vtkRenderingMatplotlib
     vtkCommonDataModel
     vtkCommonTransforms
     vtkImagingCore
-    vtkPython
     vtkRenderingCore
     vtkWrappingPythonCore
+  COMPILE_DEPENDS
+    vtkPython
+  OPTIONAL_PYTHON_LINK
   )
diff --git a/Utilities/PythonInterpreter/module.cmake b/Utilities/PythonInterpreter/module.cmake
index 3dc2b892..a6cf5a6d 100644
--- a/Utilities/PythonInterpreter/module.cmake
+++ b/Utilities/PythonInterpreter/module.cmake
@@ -1,8 +1,9 @@
 vtk_module(vtkPythonInterpreter
   DEPENDS
     vtkCommonCore
-    vtkPython
     vtksys
+  COMPILE_DEPENDS
+    vtkPython
   EXCLUDE_FROM_WRAPPING
   PRIVATE_DEPENDS
     vtkCommonMisc
