commit 62dc757d6f8f541da67d54cf24c621b7c7ef17a4
Author: looooo <sppedflyer@gmail.com>
Date:   Thu Jan 31 11:15:56 2019 +0100

    Prefer-dumpversion-over-version.patch

diff --git a/CMake/VTKGenerateExportHeader.cmake b/CMake/VTKGenerateExportHeader.cmake
index 9a7a7638..13fe0623 100644
--- a/CMake/VTKGenerateExportHeader.cmake
+++ b/CMake/VTKGenerateExportHeader.cmake
@@ -171,7 +171,7 @@ endmacro()
 macro(_vtk_test_compiler_hidden_visibility)
 
   if(CMAKE_COMPILER_IS_GNUCXX)
-    execute_process(COMMAND ${CMAKE_C_COMPILER} --version
+    execute_process(COMMAND ${CMAKE_C_COMPILER} --dumpversion
       OUTPUT_VARIABLE _gcc_version_info
       ERROR_VARIABLE _gcc_version_info)
     string(REGEX MATCH "[3-9]\\.[0-9]\\.[0-9]*"
