From 1bb8275a2343cbba78047d52a35be2736eba1605 Mon Sep 17 00:00:00 2001
From: Korijn van Golen <korijn@gmail.com>
Date: Fri, 1 Jul 2016 08:23:21 +0200
Subject: [PATCH] Removed unused (and correct by default) install paths

---
 recipe/bld.bat | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/recipe/bld.bat b/recipe/bld.bat
index 7679ba4..174024d 100644
--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -20,10 +20,6 @@ cmake .. -G "NMake Makefiles" ^
     -DVTK_WRAP_PYTHON:BOOL=ON ^
     -DVTK_PYTHON_VERSION:STRING="%PY_VER%" ^
     -DVTK_INSTALL_PYTHON_MODULE_DIR:PATH="%SP_DIR%" ^
-    -DINSTALL_BIN_DIR:PATH="%LIBRARY_BIN%" ^
-    -DINSTALL_LIB_DIR:PATH="%LIBRARY_LIB%" ^
-    -DINSTALL_INC_DIR:PATH="%LIBRARY_INC%" ^
-    -DINSTALL_MAN_DIR:PATH="%LIBRARY_PREFIX%/man" ^
     -DVTK_USE_SYSTEM_LIBRARIES:BOOL=ON
 if errorlevel 1 exit 1
 
-- 
1.9.4.msysgit.1

