diff --git a/Wrapping/Python/vtkmodules/generate_pyi.py b/Wrapping/Python/vtkmodules/generate_pyi.py
index fb324ff918..0c2018be04 100755
--- a/Wrapping/Python/vtkmodules/generate_pyi.py
+++ b/Wrapping/Python/vtkmodules/generate_pyi.py
@@ -47,6 +47,11 @@ import builtins
 import inspect
 import importlib.util
 
+# ==== Cancel any module overrides ====
+
+import vtkmodules
+
+vtkmodules.MODULE_MAPPER = {}
 
 # ==== For type inspection ====
 
