diff --git a/ThirdParty/glew/vtkglew/src/glew.c b/ThirdParty/glew/vtkglew/src/glew.c
index 9551175ee9..234bfed48f 100644
--- a/ThirdParty/glew/vtkglew/src/glew.c
+++ b/ThirdParty/glew/vtkglew/src/glew.c
@@ -38,7 +38,15 @@
 
 #if defined(GLEW_OSMESA)
 #  define GLAPI extern
+#  ifndef APIENTRY
+#    define APIENTRY
+#    define GLEW_APIENTRY_DEFINED
+#  endif
 #  include <GL/osmesa.h>
+#  ifdef GLEW_APIENTRY_DEFINED
+#    undef APIENTRY
+#    undef GLEW_APIENTRY_DEFINED
+#  endif
 #elif defined(GLEW_EGL)
 #  include <GL/eglew.h>
 #elif defined(_WIN32)
