From 172e85feaf7cd36533012d66fb12a8942888d61d Mon Sep 17 00:00:00 2001
From: Ben Boeckel <ben.boeckel@kitware.com>
Date: Fri, 19 Feb 2021 12:41:29 -0500
Subject: [PATCH] FindTBB: chain off to TBBConfig if available

Fixes: #18112
---
 CMake/FindTBB.cmake | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/CMake/FindTBB.cmake b/CMake/FindTBB.cmake
index 27576a5cc8..5ba89d52fb 100644
--- a/CMake/FindTBB.cmake
+++ b/CMake/FindTBB.cmake
@@ -68,6 +68,32 @@
 #  FindTBB helper functions and macros
 #
 
+# Use TBBConfig.cmake if possible.
+
+set(_tbb_find_quiet)
+if (TBB_FIND_QUIETLY)
+  set(_tbb_find_quiet QUIET)
+endif ()
+set(_tbb_find_components)
+set(_tbb_find_optional_components)
+foreach (_tbb_find_component IN LISTS TBB_FIND_COMPONENTS)
+  if (TBB_FIND_REQUIRED_${_tbb_find_component})
+    list(APPEND _tbb_find_components "${_tbb_find_component}")
+  else ()
+    list(APPEND _tbb_find_optional_components "${_tbb_find_component}")
+  endif ()
+endforeach ()
+unset(_tbb_find_component)
+find_package(TBB CONFIG ${_tbb_find_quiet}
+  COMPONENTS ${_tbb_find_components}
+  OPTIONAL_COMPONENTS ${_tbb_find_optional_components})
+unset(_tbb_find_quiet)
+unset(_tbb_find_components)
+unset(_tbb_find_optional_components)
+if (TBB_FOUND)
+  return ()
+endif ()
+
 #====================================================
 # Fix the library path in case it is a linker script
 #====================================================
-- 
2.33.0

