diff --git a/ThirdParty/ioss/vtkioss/Ioss_Decomposition.C b/ThirdParty/ioss/vtkioss/Ioss_Decomposition.C
index a3f69e7fc4c..1cbf2c62295 100644
--- a/ThirdParty/ioss/vtkioss/Ioss_Decomposition.C
+++ b/ThirdParty/ioss/vtkioss/Ioss_Decomposition.C
@@ -16,6 +16,7 @@
 #include <cassert>
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #include <numeric>
 
 #if !defined(NO_ZOLTAN_SUPPORT)
diff --git a/ThirdParty/ioss/vtkioss/Ioss_ParallelUtils.h b/ThirdParty/ioss/vtkioss/Ioss_ParallelUtils.h
index dce6a2005fb..7e1b9f40547 100644
--- a/ThirdParty/ioss/vtkioss/Ioss_ParallelUtils.h
+++ b/ThirdParty/ioss/vtkioss/Ioss_ParallelUtils.h
@@ -20,6 +20,7 @@
 #include "vtk_fmt.h"
 #include VTK_FMT(fmt/format.h)
 #include VTK_FMT(fmt/ostream.h)
+#include VTK_FMT(fmt/ranges.h)
 #endif
 
 #ifdef SEACAS_HAVE_MPI
